nucypher/datastore/datastore.py

```diff
@@ -14,269 +14,222 @@
+        where the `Datastore` will _always_ return specific keys, but queries
+        will almost always be partial keys.
         """
-        Adds a Workorder to the keystore.
-        """
-        session = session or self._session_on_init_thread
-
-        # Get or Create Bob Verifying Key
-        fingerprint = fingerprint_from_key(bob_verifying_key)
-        key = session.query(Key).filter_by(fingerprint=fingerprint).first()
-        if not key:
-            key = self.add_key(key=bob_verifying_key)
-
-        new_workorder = Workorder(bob_verifying_key_id=key.id,
-                                  bob_signature=bob_signature,
-                                  arrangement_id=arrangement_id)
-
-        session.add(new_workorder)
-        self.__commit(session=session)
-        return new_workorder
-
-    def get_workorders(self,
-                       arrangement_id: bytes = None,
-                       bob_verifying_key: bytes = None,
-                       session=None
-                       ) -> List[Workorder]:
-        """
-        Returns a list of Workorders by HRAC.
-        """
-        session = session or self._session_on_init_thread
-        query = session.query(Workorder)
-
-        if not arrangement_id and not bob_verifying_key:
-            workorders = query.all()  # Return all records
-
-        else:
-
-            # Return arrangement records
-            if arrangement_id:
-                workorders = query.filter_by(arrangement_id=arrangement_id)
-
-            # Return records for Bob
-            else:
-                fingerprint = fingerprint_from_key(bob_verifying_key)
-                key = session.query(Key).filter_by(fingerprint=fingerprint).first()
-                workorders = query.filter_by(bob_verifying_key_id=key.id)
+        other_key = DatastoreKey.from_bytestring(key_bytestring)
+        return self.record_type == (other_key.record_type or self.record_type) and \
+               self.record_field == (other_key.record_field or self.record_field) and \
+               self.record_id == (other_key.record_id or self.record_id)
 
-            if not workorders:
-                raise NotFound
 
-        return list(workorders)
-
-    def del_workorders(self, arrangement_id: bytes, session=None) -> int:
-        """
-        Deletes a Workorder from the Keystore.
-        """
-        session = session or self._session_on_init_thread
+class Datastore:
+    """
+    A persistent storage layer for arbitrary data for use by NuCypher characters.
+    """
 
-        workorders = session.query(Workorder).filter_by(arrangement_id=arrangement_id)
-        deleted = workorders.delete()
-        self.__commit(session=session)
-        return deleted
+    # LMDB has a `map_size` arg that caps the total size of the database.
+    # We can set this arbitrarily high (1TB) to prevent any run-time crashes.
+    LMDB_MAP_SIZE = 1_000_000_000_000
```

Comment:
We've recently observed this value leading to the inability to allocate enough vram to run the tests that include concurrent ursulas and multiple lmdb instances.  

Let's set a different value for tests vs. production and go from there.

Reply:
I think I resolved this with a monkeypatch. Will you take a peek in the latest commit?